content/blog/2023/07/22/2023-07-22-gsoc-2023-midterm.adoc

```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title: Jenkins in Google Summer of Code Midterm Recap
+tags:
+- gsoc
+- gsoc2023
+- midterm
+authors:
+- ash-sxn
+- harsh-ps-2003
+- jagruti
+- vandit1604
+opengraph:
+  image: /images/gsoc/opengraph.png
+sig:
+- gsoc
+---
+
+The Jenkins in GSoC program recently reached its midterm milestone for GSoC 2023.
+This milestone served to assess performance, and precipitate pass/fail decisions.
+Taking time to evaluate the progress and workflow of the project provides an opportunity to correct course and address underlying issues.
+
+Earlier this month, on the link:https://www.meetup.com/jenkins-online-meetup/[Jenkins Online Meetup], GSoC contributors presented midterm status and progress for their projects, below is a recap.
+Thank you Jagruti, Harsh, Vandit, and Ashutosh for your contributions to Jenkins!
+
+== link:https://www.jenkins.io/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Add Probes to "Plugin Health Score"]
+By link:https://github.com/Jagrutiti/[Jagruti Tiwari]
+
+A probe collects data about each plugin.
+It is an automated system ensuring that the Jenkins Community defined rules are followed.
+Probe data is used by health scoring to compute a score.
+A higher quality probe is given a good score.
+An obsolete or outdated plugin is given a lower score.
+
+Sharing details about the probes that I worked on during the first phase of GSoC.
+
+=== Unreleased Production Changes Probe
+
+A bug fix, security fix, or new features are of no use unless they are released.
+The "unreleased production changes" probe looks for unreleased commits in the production files.
+
+The probe aims to identify unreleased production commits in the Maven production files.
+
+My mentors suggested I should start with writing the test cases to reproduce the probe requirements, and later add code to pass each of the test cases.
+
+I implemented the JGit library and understood its features.
+I lacked depth in Git concepts that made me struggle with JGit.
+This led me to strengthen the Git concepts along the way.
+
+=== The Third-Party Repository Detection Probe
+
+By definition, a third-party repository is a repository that is not hosted at link:https://repo.jenkins-ci.org/[].
+
+The third-party repository detection probe ensures that plugins are built only from trusted and reliable Maven repositories.
+
+This probe was requested by the community.
+
+A major challenge in implementing this probe was considering the parent and child pom relationship from all angles.
+
+Some edge-case scenarios were not considered while specifying the probe.
+And, due to the number of open questions, this probe was put on hold until further research was completed.
+
+In the process, I have extensively read the Maven documentation to understand how to test POM structure, inheritance, and Maven API.
+
```

Comment:
Read the link:blog/2023/07/20/third-party-repository-detection-probe.adoc[unreleased production changes] blog post for more details on the probe.